Celeste.Mod.mm/Patches/EntityData.cs

```diff
@@ -12,5 +12,19 @@ class patch_EntityData : EntityData {
             return orig_Has(key);
         }
 
+        /// <summary>
+        /// Get the <see cref="T:System.String" /> value associated with a key.
+        /// An Empty or pure Whitespace value will result in the <paramref name="defaultValue"/> to be returned.
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="defaultValue"></param>
+        /// <returns></returns>
+        public string String(string key, string defaultValue = null) {
+            if (Values != null && Values.TryGetValue(key, out object value)
+                && value != null && value.ToString().Trim().Length > 0) {
```

Comment:
`value != null && value.ToString().Trim().Length > 0` can literally just be `!string.IsNullOrWhileSpace(value?.ToString())`, which I feel convays intent more clearly.
Also it might reduce a string allocation by not calling `Trim()`

Reply:
It is simpler but no string allocation simplicity because that's literally what IsNullOrWhiteSpace does to check white space

Reply:
`string.IsNullOrWhiteSpace` loops over the string and does not allocate.
Also did you intend on using `string.IsNullOrEmpty`?

Reply:
No because essentially empty strings but only white space strings should be considered empty if you're trying to get a string out, I believe

Reply:
..I don't understand what you're trying to say? 
White-space only strings should fall back to the default value, right? In that case `string.IsNullOrEmpty` doesn't cover that case, but `string.IsNullOrWhiteSpace` does.

Reply:
That's exactly what I said but phased way better